ext/storage_sources/s3_store/aws_bucket_conn.h

```diff
@@ -0,0 +1,30 @@
+
+#ifndef AWS_BUCKET_CONN
+#define AWS_BUCKET_CONN
+
+#include <aws/core/Aws.h>
+#include <aws/s3-crt/S3CrtClient.h>
+#include <aws/s3-crt/model/Object.h>
+
+#include <string>
+#include <vector>
+
+/*
+ * Singleton class owning the database connection, provides access to sessions and any other
+ * required connection API calls.
+ */
+class aws_bucket_conn {
+    public:
+    explicit aws_bucket_conn(const Aws::S3Crt::ClientConfiguration &config);
+    bool list_buckets(std::vector<std::string> &buckets) const;
+    bool list_objects(
+      const std::string bucket_name, std::vector<Aws::S3Crt::Model::Object> &bucket_objects) const;
+    bool put_object(const Aws::String &bucket_name, const Aws::String &object_key,
+      const Aws::String &file_name) const;
+    bool delete_object(const Aws::String &bucket_name, const Aws::String &object_key) const;
+    ~aws_bucket_conn();
```

Comment:
I don't think we need to define a destructor if there is no specific behavior. However, it can be considered good practice to define it even if it does nothing. In this case, we can make it equal to `default` and remove the body from the `cpp` file.